include/grpc++/support/string_ref.h

```diff
@@ -50,22 +50,22 @@ class string_ref {
   typedef std::reverse_iterator<const_iterator> const_reverse_iterator;
 
   // constants
-  static constexpr size_t npos = size_t(-1);
+  const static size_t npos = size_t(-1);
```

Comment:
Maybe add some language that we're deviating from the proposed standard because of old compilers here (see what I did for `compare` for example. This could be a class comment too (as opposed to individual functions.